src/map/packets/message_special.h

```diff
@@ -28,6 +28,11 @@
 
 #include "basic.h"
 
+enum MSGSPECIAL_ID : uint16
+{
+    MSGSPECIAL_KEYITEM_OBTAINED = 6390, /* Obtained key item: <item>. */
```

Comment:
These text IDs are per zone. Gonna have to figure out a different way to handle this. Maybe a helper that calls to npcUtil.giveKeyItem? Not sure the correct way to handle C++ -> Lua.

Reply:
This would be a good situation to use `luautils::GetTextIDVariable`, but it'll need a bit of work since it wasn't updated with the tabling of IDs.

Reply:
_suddenly remebers the textids used for abyssea cruor_

😅 Don't mind me

Reply:
Ah ok, thanks for letting me know. This technically is always in the MH but I'll look at merging in current master and trying to use this mentioned `luautils::GetTextIDVariable().